src/unix/udp.c

```diff
@@ -230,6 +227,15 @@ static int uv__udp_recvmmsg(uv_udp_t* handle, uv_buf_t* buf) {
                       msgs[k].msg_hdr.msg_name,
                       flags);
     }
+
+    if (handle->recv_cb != NULL) {
+      chunk_buf = uv_buf_init(iov[0].iov_base, iov[0].iov_len);
+      handle->recv_cb(handle,
+                      msgs[0].msg_len,
+                      &chunk_buf,
+                      msgs[0].msg_hdr.msg_name,
+                      UV_UDP_MMSG_FREE_BUF);
```

Comment:
I must be missing something very obvious here, but does this mean that you are receiving the first packet twice now (once with `UV_UDP_MMSG_CHUNK` and once with `UV_UDP_MMSG_FREE_BUF`?

Reply:
That's the way to tell the app that it can now free the entire buffer. Probably nread could be 0 on this last recv_cb. And `UV_UDP_MMSG_FREE_BUF` could just not exist. The app does not need it in order to know that it has to free the buffer.

Reply:
Correct, the first chunk is sent both as a regular chunk and at the end with the flag which just indicates “don’t process this, just free the buffer”. 

That’s the proposal.